magenta-lib/src/main/scala/magenta/input/DeploymentResolver.scala

```diff
@@ -0,0 +1,68 @@
+package magenta.input
+
+import play.api.libs.json.JsValue
+
+case class Deployment(
+  name: String,
+  `type`: String,
+  stacks: List[String],
+  regions: List[String],
+  dependencies: List[String],
+  app: String,
+  contentDirectory: String,
+  parameters: Map[String, JsValue]
+)
+
+object DeploymentResolver {
+  val DEFAULT_REGIONS = List("eu-west-1")
+
+  def resolveDeployment(name: String, deployment: DeploymentOrTemplate, yaml: RiffRaffYaml): Either[List[String], DeploymentOrTemplate] = {
+    val missingDeps = yaml.missingDependencies(deployment.dependencies.getOrElse(Nil))
+    if (missingDeps.nonEmpty)
+      Left(missingDeps.map(dep => s"Missing dependency $dep in $name"))
+    else {
+      deployment match {
+        case deployment @ DeploymentOrTemplate(Some(deploymentType), None, stacks, regions, _, _, _, _) =>
+          // terminating case - apply any default stacks and regions
+          val deploymentWithGlobalDefaults = deployment.copy(stacks = stacks.orElse(yaml.stacks), regions = regions.orElse(yaml.regions))
+          Right(deploymentWithGlobalDefaults)
+
+        case template @ DeploymentOrTemplate(None, Some(templateName), stacks, regions, app, contentDirectory, dependencies, parameters) =>
+          // template case - recursively resolve
+          for {
+            template <- yaml.templates.flatMap(_.get(templateName))
+              .toRight(List(s"Template with name $templateName (specified in $name) does not exist")).right
```

Comment:
Agreed. I like the change in your PR - although wonder if it still needs to be a list.